File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    
    dicti={}
    def climbStairs(self, n: int) -> int:
        
        if(n==1):
                return 1
        if n==2:
                return 2
        
        if n in self.dicti:
            return self.dicti[n]
        
        
        self.dicti[n]=self.climbStairs(n-1)+self.climbStairs(n-2)
        return self.dicti[n]

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
           int a=1;
           int b=2;
        
            if(n<=2){
                    return n;
            }
            int ans=0;
            for(int i=3;i<=n;i++){
                    
                    ans=a+b;
                    a=b;
                    b=ans;
                    
                
                
            }
        
            return ans;
    }
};